strawberry/extensions/query_complexity_estimator.py

```diff
@@ -0,0 +1,375 @@
+            name: self._estimate_cost(
+                parent_type=RootTypeInfo.VALUE,
+                node=query,
+                fragments=fragments,
+            )
+            for name, query in queries.items()
+        }
+
+        self.results = query_complexities
+
+        if callable(self.callback):
+            self.callback(query_complexities, self.execution_context)
+
+        yield
+
+    def _get_type_info(self, parent_type_name: str, field_name: str) -> TypeInfo:
+        schema = self.execution_context.schema
+        strawberry_field = schema.get_field_for_type(field_name, parent_type_name)
+        assert strawberry_field is not None
+
+        field_type = strawberry_field.type_annotation.resolve()
+        if isinstance(field_type, StrawberryContainer):
+            field_type_name = field_type.of_type.__name__
+        else:
+            field_type_name = field_type.__name__
+
+        field_estimators = [
+            e
+            for e in strawberry_field.extensions
+            if isinstance(e, FieldComplexityEstimator)
+        ]
+        field_estimator = field_estimators[0] if len(field_estimators) > 0 else None
+
+        return NonRootTypeInfo(
+            parent_type_name=parent_type_name,
+            name=field_type_name,
+            source_field=strawberry_field,
+            estimator=field_estimator,
+        )
+
+    def _child_complexities_lazy(
+        self,
+        node: Union[FieldNode, FragmentDefinitionNode],
+        node_type: TypeInfo,
+        fragments: Dict[str, FragmentDefinitionNode],
+    ) -> Iterator[int]:
+        """A lazy generator over the complexities of the children of a node.
+
+        Instead of eagerly evaluating child complexities, we pass a lazy generator to the
+        evaluator. The advantage with this is that the evaluator can decide to just
+        ignore child costs, and then we don't waste time calculating them
+        """
+        if node.selection_set is None:
+            return
+
+        for child in node.selection_set.selections:
+            yield self._estimate_cost(
+                parent_type=node_type,
+                node=child,
+                fragments=fragments,
+            )
+
+    def _estimate_cost(
+        self,
+        parent_type: TypeInfo,
+        node: Node,
+        fragments: Dict[str, FragmentDefinitionNode],
+    ) -> int:
+        if isinstance(node, OperationDefinitionNode):
+            return sum(
+                self._estimate_cost(
+                    parent_type=parent_type, node=child, fragments=fragments
+                )
+                for child in node.selection_set.selections
+            )
+
+        if isinstance(node, FieldNode):
+            node_type = self._get_type_info(parent_type.name, node.name.value)
+            variables = self.execution_context.variables or {}
+            node_body = node
+            args = {
+                to_snake_case(arg.name.value): variables.get(arg.value.name.value, None)
```

Comment:
Is there an inverse function to `self.execution_context.schema.config.name_converter.apply_naming_config` that will turn camel into snake case without hardcoding it here. What we need is to get the `python_name` of a field from its `graphql_name`